### ISSUE TITLE
Unexpected CS1573 with XML docs that are already inherited

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#69126]
**Version Used**: VS 17.6.5, Roslyn 4.6.0-3.23259.8 (c3cc1d0c)

**Steps to Reproduce**:

1. Enable CS1573 in .editorconfig (ie. force all members to have XML docs)
2. Paste the following code:
```csharp
/// <summary>
/// A blah.
/// </summary>
/// <param name="Foo">This is Foo.</param>
public record A(string Foo)
{
}

/// <summary>
/// B blah.
/// </summary>
/// <param name="Bar">This is Bar.</param>
/// <inheritdoc cref="A"/>
public record B(string Foo, string Bar) : A(Foo)
{
}
``` 

**Diagnostic Id**: CS1573

**Expected Behavior**: no warnings/errors should show up, I'd expect `inheritdoc` to just "fill in" any missing XML docs from parents.

**Actual Behavior**: you get a CS1573 saying `Foo` has no matching XML docs, even though it's inheriting them from `A`, and you can even see those docs literally showing up just fine in the quick info popup (??) 😶

![image](https://github.com/dotnet/roslyn/assets/10199417/a1d8c85b-f1d6-499f-a2bd-a46d96ea5729)

**Comment by jcouv** [comment]
I think this behavior is by-design. There is an inheritance relationship between `A` and `B`, but there is not inheritance relationship between the parameter lists on `A` and `B` primary constructors. The parameter names and types could be different.
Tagging @sharwell in case anything to add.
